GUIDELINES.md

```diff
@@ -4,36 +4,58 @@ Note: This is a living document managed by the SDL Project Maintainer and may ch
 - All `set` methods should return `this` in order to support method chaining.
-- Classes must contain a constructor, but the constructor _should not_ contain parameter type/value checking as a workaround for method overloading. For example, a semantic versioning class should have a constructor which accepts 3 parameters (major, minor, patch) and assigned to properties of the same/similar names prefixed with `_`. To also allow a semantic version string (e.g. "2.1.3") as input, a new `fromString(version)` method should be added to the class definition which parses the string and sets the proper `_major`, `_minor`, and `_patch` properties. See `/lib/js/util/Version.js` in the repository for an example.
+- Classes must contain a constructor, but the constructor _should not_ contain parameter type/value checking as a workaround for method overloading. For example, a semantic versioning class should have a constructor which accepts 3 parameters (major, minor, patch) and assigned to properties of the same/similar names prefixed with `_`. To also allow a semantic version string (e.g. "2.1.3") as input, a new `fromString(version)` method should be added to the class definition which parses the string and sets the proper `_major`, `_minor`, and `_patch` properties. See `/lib/js/src/util/Version.js` in the repository for an example.
 - Constructor parameters should be optional in most cases to support utilizing class methods as a helper without the need to initialize an instance with valid parameters.
 
 
 ### Enumerations
-Since JavaScript does not offer traditional enumerations, an abstract `Enum` class has been defined (see `/lib/js/util/Enum.js`) which shall be extended for the purpose of creating new enumerations. Each extended enumeration class must be named with the suffix `Type`, contain a constructor which calls `super()`, use ES6-style getters as `static` methods to support simple dot-access notation, and contain a `static valueForString(value)` method which returns a given enumeration **value** (not key) if it exists in the collection (otherwise `null`). See `/lib/js/transport/enums/TransportType.js` for an example.
+Since JavaScript does not offer traditional enumerations, an abstract `Enum` class has been defined (see `/lib/js/src/util/Enum.js`) which shall be extended for the purpose of creating new enumerations. Each extended enumeration class must be named to match the enum name in the RPC Spec, contain a constructor which calls `super()`, and contain a `static valueForString(key)` method which returns a given enumeration **value** if the provided **key** exists in the collection (otherwise `null`) using the `_valueForString(key)` private method found in the base `Enum` class. See `/lib/js/src/transport/enums/TransportType.js` for an example.
```

Comment:
I think the call out to the RPC spec should be separated from this section and put into the RPC generation section. Maybe leave a sentence here like `Special rules exist for enums contained in the RPC spec, please see the section RPC Class Generation`.